src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -204,9 +207,13 @@ export class AccountManagementService implements IAccountManagementService {
 
 		return this.doWithProvider(account.key.providerId, async (provider) => {
 			let refreshedAccount = await provider.provider.refresh(account);
-			if (self.isCanceledResult(refreshedAccount)) {
-				// Pattern here is to throw if this fails. Handled upstream.
-				throw new Error(localize('refreshFailed', "Refresh account was canceled by the user"));
+			if (self.isPromptFailedResult(refreshedAccount)) {
+				if (refreshedAccount.canceled) {
+					// Pattern here is to throw if this fails. Handled upstream.
+					throw new Error(localize('refreshFailed', "Refresh account was canceled by the user"));
+				} else {
+					throw Error(localize('refreshFailed', `${refreshedAccount.errorCode} \nError Message: ${refreshedAccount.errorMessage}`));
```

Comment:
nit: While technically fine - it would be good to be consistent with how we normally construct instances of classes and use `new`